### PR TITLE
Add result arbitration to WASM task

### PR DESCRIPTION
**Pull request by mplebanski**

### CODE REVIEW

File: apps/wasm/task.py [hunk 1]
```diff
@@ -111,6 +111,12 @@ def get_verdicts(self):
 
         return verdicts
 
+    def get_subtask_count(self) -> int:
+        instances_cnt = len(self.get_instances())
```
Comment (mplebanski) [inline apps/wasm/task.py:119]: Add a comment here

File: apps/wasm/task.py [hunk 2]
```diff
@@ -397,18 +408,18 @@ def verify_task(self):
         return self.finished_computation()
 
     def get_total_tasks(self):
-        return (WasmTask.REDUNDANCY_FACTOR + 1) * len(self.subtasks)
+        total = sum([s.get_subtask_count() for s in self.subtasks])
+        logger.info("Total tasks: %d", total)
```
Comment (mplebanski) [inline apps/wasm/task.py]: Remove this log

File: apps/wasm/task.py [hunk 3]
```diff
@@ -436,21 +447,18 @@ def abort(self):
...
             return 0.0
 
-        num_finished = len(list(filter(lambda x: x.is_finished(),
-                                       self.subtasks)))
+        tasks_left = self.get_tasks_left()
+
+        assert num_total >= tasks_left
+
+        progress = (num_total - tasks_left) / num_total
+        logger.info("progress: %d", progress)
```
Comment (mplebanski) [inline apps/wasm/task.py]: Remove this log

File: apps/wasm/task.py [hunk 4]
```diff
@@ -397,18 +408,18 @@ def verify_task(self):
         return self.finished_computation()
 
     def get_total_tasks(self):
-        return (WasmTask.REDUNDANCY_FACTOR + 1) * len(self.subtasks)
+        total = sum([s.get_subtask_count() for s in self.subtasks])
+        logger.info("Total tasks: %d", total)
+        return total
 
     def get_active_tasks(self):
```
Comment (mplebanski) [inline apps/wasm/task.py:423]: Improve this method

Comment (mbenke) [reply apps/wasm/task.py:423]: This potentially influences progress bar, but after discussion with @lukasz-glen we have concluded, we can live with it.

**Review by mbenke — APPROVED** [review]

**Pull request merged**